### PR TITLE
[Policies] Adding S3 policies report

### PR DESCRIPTION
**Pull request by jgchoppe**
This PR is to add s3 policies report to the s3-acl-viewer tools. In fact, you'll have the possibility to have a report about policies used by s3 buckets for an account.
Here is an example:
s3_policies

### CODE REVIEW

File: s3.py [hunk 1]
```diff
@@ -64,12 +68,56 @@ def dump_gspread(self):
         return self.dump_csv()
 
 
+def get_format_principal(policy):
+    for statement in policy["Statement"]:
+        try:
+            statement["Principal"]["AWS"] = statement["Principal"]["AWS"] \
+                if type(statement["Principal"]["AWS"]) is str else ', '.join(statement["Principal"]["AWS"])
```
Comment (thibautcornolti) [inline s3.py]: You should use a function for this

File: s3.py [hunk 2]
```diff
@@ -64,12 +68,56 @@ def dump_gspread(self):
         return self.dump_csv()
 
 
+def get_format_principal(policy):
+    for statement in policy["Statement"]:
+        try:
```
Comment (thibautcornolti) [inline s3.py]: Use a `for` to test all keys

File: build_xlsx.py [hunk 3]
```diff
@@ -60,6 +61,124 @@ def write_data(workbook, worksheet, buckets):
...
+    })
+    worksheet.conditional_format(1, 1, line, 11, {
+        "type":     "text",
+        "criteria": "containing",
+        "value":    "Allow",
+        "format":   red,
+    })
+
+def write_right_principal(worksheet, policies, data_format, i, color_format):
+    try:
```
Comment (thibautcornolti) [inline build_xlsx.py]: You should avoid chaining try / catch like this

**Review by thibautcornolti — CHANGES_REQUESTED** [review]

**Review by thibautcornolti — APPROVED** [review]

File: s3.py (hunk 1 above)
Comment (jgchoppe) [reply s3.py]: fixed

File: s3.py (hunk 2 above)
Comment (jgchoppe) [reply s3.py]: fixed

File: build_xlsx.py (hunk 3 above)
Comment (jgchoppe) [reply build_xlsx.py]: fixed

**Pull request merged**